README.md

```diff
@@ -94,40 +94,47 @@ Otherwise, follow the steps above. The 12B param model may not function well in
 ### Training on Other Instances
 
 A100 instance types are not available in all cloud regions, or can be hard to provision. Training is possible on other GPU instance types, 
-for smaller Dolly model sizes, and with small modifications to reduce memory usage.
-These modifications are not optimal, but are simple to make.
+for smaller Dolly model sizes, and with small modifications to reduce memory usage. These modifications are not optimal, but are simple to make. 
+
+Select your GPU family type from the `gpu_family` widget, enter the number of GPUs available in the `num_gpus` widget, and then run the rest of the code. 
+A number of different options will be set for you to train the model for one of the following GPU types:
+- A100 (default)
+- A10 
+- V100
+
+Details of the different configurations are below.
+
+#### A100 GPUs
+
+A100 GPUs are preferred for training all model sizes, and are the only GPUs that can train the 12B param model in a reasonable amount of time.
+As such, this is the default configuration, as set in the `a100_config.json` deepspeed config file.
 
 #### A10 GPUs
 
 Training the 12B param model is not recommended on A10s.
 
-To train the 6.9B param model on A10 instances (ex: `g5.24xlarge`, 4 x A10 24GB; `Standard_NV72ads_A10_v5`, 2 x A10), make the following changes:
+To train the 6.9B param model on A10 instances (ex: `g5.24xlarge`, 4 x A10 24GB; `Standard_NV72ads_A10_v5`, 2 x A10),
+simply select `a10` from the `gpu_family` widget and enter the number of GPUs available in the `num_gpus` widget, then run the rest of the code. 
+This will use the `a10_config.json` deepspeed config file, which makes the following changes:
 
-- Set `per-device-train-batch-size` and `per-device-eval-batch-size` to 3 in the `train_dolly.py` invocation of `deepspeed`
-- Modify the deepspeed config file `ds_z3_bf16_config.json` to configure optimizer offload. Within the `"zero_optimization"` section, add:
+- `per-device-train-batch-size` and `per-device-eval-batch-size` are set to 3 in the `train_dolly.py` invocation of `deepspeed`
+- Within the `"zero_optimization"` section of the deepspeed config, we have added:
   ```
   "offload_optimizer": {
     "device": "cpu",
     "pin_memory": true
   },
   ```
-- Set the `num_gpus` widget in `train_dolly` to the number of GPUs in your instance, such as 2 or 4, before running
-
-To train the 2.8B param model:
-
-- Instead, only set `per-device-train-batch-size` and `per-device-eval-batch-size` to 3 in the `train_dolly.py` invocation of `deepspeed`
```

Comment:
The only issue I see is that optimizer offload is enabled even when training the 3B model now, when it's not necessary and would slow it down. Hm. We could just ignore that - it'll work just slower. I don't love making a second config file variation for it. IT's possible instead to increase batch size with param offload and 3B. Opinions?

Reply:
Hmmm - not sure. Let me test a couple variations and see which performs better.

Reply:
Ok, I'm running 2 test versions now. Both MLR 13.0 on g5.12xlarge instances to train the pythian-2.8B model.

1. Removed the optimizer offload section of the A10 config, left everything else the same (batch size = 3)
2. Left in the optimizer offload section, set batch size = 6

Let's see which one finishes faster...

Reply:
Hmmm - ok, well test case 1. failed with:
```
File "/databricks/python/lib/python3.10/site-packages/torch/autograd/__init__.py", line 197, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 434.00 MiB (GPU 3; 22.20 GiB total capacity; 19.85 GiB already allocated; 86.12 MiB free; 20.81 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
```

I'll try setting `os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "max_split_size_mb:64"`

Reply:
Same error with test 2. Seems like the optimizer offload was helping here.

Reply:
That setting really isn't the issue :) You're just OOM, and a smaller batch size is needed

Reply:
Test 1 (no optimizer offload), still failing with `max_split_size_mb` = 64, trying it at 32

Reply:
It's not going to help if you're actually OOM. I don't think this is fragmentation. You need smaller batches

Reply:
I can test with smaller batch sizes. Fwiw, test 2 has not had this error since I set the `max_split_size_mb` (but it is still using optimizer offload)

Reply:
Your other test has optimizer offload though so can use larger batch sizes without OOM. I'm just saying I don't think this setting is ever the remedy for OOM

Reply:
Yes and have 3 deepspeed configs, seems OK